File: 백준/Silver/10799. 쇠막대기/쇠막대기.py
path = input()
stack = []
answer = 0

# () -> 레이져, (()()) -> 막대( 레이져 ()  레이져 ()  )막대
for i in range (len(path)):
  if path[i] == '(':
    stack.append(path[i])
  else :
    # 현재 쇠막대기 카운팅
    if path[i-1] == '(':
      stack.pop()
      answer += len(stack)
    # 레이저 카운트
    else :
      stack.pop()
      answer += 1
print(answer)

File: 백준/Silver/10799. 쇠막대기/쇠막대기.java
import java.io.*;
import java.util.*;

public class Main {
    static int answer = 0;

    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        ArrayDeque<String> stack = new ArrayDeque<>();
        String s = br.readLine();
        for(int i = 0; i < s.length(); i++) {
            char c = s.charAt(i);
            if(c == '(') {
                stack.push(s);
            }else{
                stack.pop();
                if (s.charAt(i-1) == '(') {
                    answer += stack.size();
                }else{
                    answer += 1;
                }
            }
        }
        System.out.println(answer);

    }

}